build_tools/buildkite/check_user_auth.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import sys
+
+AUTHORIZED_USER_EMAILS = ["gcmn@google.com", "marius.brehler@iml.fraunhofer.de"]
+CREATOR_EMAIL_ENV_VAR = "BUILDKITE_BUILD_CREATOR_EMAIL"
+
+
+def main():
+  creator_email = os.getenv(CREATOR_EMAIL_ENV_VAR)
+  if not creator_email:
+    raise RuntimeError(f"{CREATOR_EMAIL_ENV_VAR} is not defined")
+
+  if creator_email.endswith("@google.com"):
+    print(
+        f"User '{creator_email}' is authorized because email ends in '@google.com'"
```

Comment:
Did some testing on my own and confirmed with support. In the case of GitHub PR events, the creator field can't be spoofed unless someone has compromised GitHub itself. There *is* a case where it can be spoofed if it's a branch push event in the repository itself, then Buildkite just uses information from git (which is it's baseline) and then it's just committer and author, which is of course not verified at all in raw git. If someone already has write access to our repository though, then I think we're ok with them running presubmits :grin: It is something to note though.

Reply:
Thanks for checking!